src/tests/ftest/launch.py

```diff
@@ -59,9 +59,9 @@
 PROVIDER_KEYS = OrderedDict(
     [
         ("cxi", "ofi+cxi"),
-        ("verbs", "ofi+verbs"),
+        ("verbs", "ofi+verbs;ofi_rxm"),
         ("ucx", "ucx+dc_x"),
-        ("tcp", "ofi+tcp"),
+        ("tcp", "ofi+tcp;ofi_rxm"),
```

Comment:
Hm, for the record the Jenkinsfile is different in the master branch, and the defaultValue string for TestProvider is empty.

Reply:
Yeah, the TCP provider testing is a different branch / different Jenkinsfile. That makes updating difficult. Because if we update one branch without updating the other, something will break.

Reply:
@phender maybe we could do this?
1. default `ofi+tcp` -> `ofi+tcp;ofi_rxm` in launch.py
2. update the provider testing branch to use `ofi+tcp;ofi_rxm`
3. remove the default `ofi+tcp` -> `ofi+tcp;ofi_rxm` in launch.py

Reply:
> @phender I think this will break TCP provider testing, right?
> 
> https://github.com/daos-stack/daos/blob/f33fae751e56e6926f69b453ec2de6d5014f0850/Jenkinsfile#L82

It shouldn't.  The `PROVIDER_KEYS.value` is used to ultimately set `CRT_PHY_ADDR_STR`, which is then used here https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/server_utils_params.py#L111 to set the default value for the server config file `provider` entry.

We may need to adjust https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/network_utils.py#L16.

We should test this PR with different `Test-provider: <proivider>` commit pragmas (or build params) and include at least `test_dmg_network_scan_basic` along with the `pr` tag.

Reply:
But if `--provider ofi+tcp` is passed, it wouldn't be in `PROVIDER_KEYS`, since `PROVIDER_KEYS` now has `ofi+tcp;ofi_rxm`, but NOT `ofi+tcp`

Reply:
We'll also need to update the descriptive messages in the Jenkinsfile, e.g. https://github.com/daos-stack/daos/blob/provider-testing-tcp/Jenkinsfile#L83, in the affected branches: `master`, `provider-testing-tcp`, release/2.4`, and `provider-testing-tcp-2.4-testing` (assuming this will be cherry-picked).

Reply:
> @phender maybe we could do this?
> 
> 1. default `ofi+tcp` -> `ofi+tcp;ofi_rxm` in launch.py
> 2. update the provider testing branch to use `ofi+tcp;ofi_rxm`
> 3. remove the default `ofi+tcp` -> `ofi+tcp;ofi_rxm` in launch.py

I'm not sure we need to go that much trouble as long as we're able to update the corresponding `provider-*-testing` branch before it runs over the weekend after the master/release branch changes land.  That said this `provider-*-testing` branch could/should be run as a parallel PR - and could serve as the verification of the tcp provider change, possibly negating the need for this PR to be run with multiple providers.

Reply:
> We should test this PR with different `Test-provider: <proivider>` commit pragmas (or build params) and include at least `test_dmg_network_scan_basic` along with the `pr` tag.

Will do. I need to re-run anyway to skip some tests that are failing due to a different bug.